src/vsr/journal.zig

```diff
@@ -725,6 +727,7 @@ pub fn Journal(comptime Replica: type, comptime Storage: type) type {
             if (self.header_with_op_and_checksum(op, checksum)) |exact| {
                 if (exact.size == @sizeOf(Header)) {
                     message.header.* = exact.*;
+                    std.mem.set(u8, message.buffer[@sizeOf(Header)..config.sector_size], 0);
```

Comment:
My guess is that it's in case this message is received locally and persisted to disk, so that we don't leak random data into the sector padding? I think usually the MessageBus would otherwise take care of zeroing the padding here?

Reply:
We could also assert this in debug builds or for `config.verify`, at call sites where we write this message to disk.

Reply:
Yeah, this isn't directly related to trust-committed-ops, it is a just a bug I ran into during testing!

It is documented in the commit https://github.com/tigerbeetledb/tigerbeetle/pull/102/commits/53e7d106e1f60751e54ae8c2685a689931807a53. But I'll add an explanation in the code, too.

> We could also assert this in debug builds or for `config.verify`, at call sites where we write this message to disk.

We do — that's what caught this error :grin:.

Reply:
Nice! I had a feeling we had those asserts in there somewhere but thought I'd pretend we didn't just in case!

Reply:
😁